src/Domain/Entities/Payment.cs

```diff
@@ -1,17 +1,35 @@
 ﻿using Domain.Common;
 using System;
+using System.ComponentModel.DataAnnotations;
 
 namespace Domain.Entities
 {
     public class Payment : BaseEntity
     {
+        [Required]
         public int OrderId { get; set; }
+
+        [Required]
         public string Method { get; set; }
+
+        [Required]
+        [RegularExpression(@"^\d+\.\d{0,2}$")]
+        [Range(0, 9999999999999999.99, ErrorMessage = "Invalid Total Amount; Max 18 digits")]
```

Comment:
Both RegExp and Range is not necessary. Use RegExp `^\d{1,18}(\.\d{1,2}){0,1}$` or the Range.